src/autoSave.js

```diff
@@ -0,0 +1,98 @@
+const { promisify } = require('util');
+const path = require('path');
+const fs = require('fs');
+
+const stat = promisify(fs.stat);
+const mkdir = promisify(fs.mkdir);
+const writeFile = promisify(fs.writeFile);
+
+function ensureDir(dirName, rt) {
+  const directoryPath = path.resolve(rt.options.baseDir, `./${dirName}`);
+  return stat(directoryPath)
+    .catch(() => mkdir(directoryPath, {
+      recursive: true,
+    })
+      .then(() => stat(directoryPath)))
+    .then((stats) => {
+      if (!stats.isDirectory()) {
+        return Promise.reject(new Error(`${directoryPath} is not a directory`));
+      }
+      return Promise.resolve(directoryPath);
+    });
+}
+
+function getComponentName(component, directoryPath) {
```

Comment:
It seems that what this function actually does is `getComponentPath()`

Reply:
Fair point, will rename

Reply:
Done